File: c/1.1.3/1.1.3_LedBarGraph.c
/**********************************************************************
* Filename    : 1.1.3_LedBarGraph.c
* Description : Make an led bar graph work.
* Author      : Robot
* E-mail      : support@sunfounder.com
* website     : www.sunfounder.com
* Update      : Cavon    2016/07/01
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>
int pins[10] = {0,1,2,3,4,5,6,8,9,10};//pin 7 is not outpin

int main(void)
{
	int i;
	if(wiringPiSetup() == -1){ //when initialize wiring failed,print messageto screen
		printf("setup wiringPi failed !");
		return 1; 
	}

	printf("\n");
	printf("========================================\n");
	printf("|             LED Bargraph             |\n");
	printf("|    ------------------------------    |\n");
	printf("|  Pin connect to #17 #18 #27 #22 #23  |\n");
	printf("|        #24 #25 #SDA #SCL #CE0        |\n");
	printf("|                                      |\n");
	printf("|  Use LED bargraph for loop iteration |\n");
	printf("|                                      |\n");
	printf("|                            SunFounder|\n");
	printf("========================================\n");
	printf("\n");

	for(i=0;i<=10;i++){       //make led pins' mode is output
		pinMode(pins[i], OUTPUT);
	}
	while(1){
		for(i=0;i<=10;i++){   //make led on from left to right
			if(i==7){continue;} //skip pin 7
			digitalWrite(i, LOW);
			delay(100);
			digitalWrite(i, HIGH);
		}
		for(i=10;i>-1;i--){   //make led on from right to left
			if(i==7){continue;}
			digitalWrite(i, LOW);
			delay(100);
			digitalWrite(i, HIGH);
		}
	}
	return 0;
}

File: python/1.1.3_LedBarGraph.py
#!/usr/bin/env python3

import RPi.GPIO as GPIO
import time

ledPins = [11, 12, 13, 15, 16, 18, 22, 3, 5, 24]

def print_msg():
	print ("========================================")
	print ("|             LED Bargraph             |")
	print ("|    ------------------------------    |")
	print ("|  Pin connect to #17 #18 #27 #22 #23  |")
	print ("|      #24 #25 #SDA #SCL #CE0          |")
	print ("|                                      |")
	print ("|  Use LED bargraph for loop iteration |")
	print ("|                                      |")
	print ("|                            SunFounder|")
	print ("========================================\n")
	print ('Program is running...')
	print ('Please press Ctrl+C to end the program...')
	input ("Press Enter to begin\n")

def setup():
	print_msg()
	GPIO.setmode(GPIO.BOARD)        # Numbers GPIOs by physical location
	for pin in ledPins:
		GPIO.setup(pin, GPIO.OUT)   # Set all ledPins' mode is output
		GPIO.output(pin, GPIO.HIGH) # Set all ledPins to high(+3.3V) to off led

def loop():
	while True:
		for pin in ledPins:		#make led on from left to right
			GPIO.output(pin, GPIO.LOW)	
			time.sleep(0.1)
			GPIO.output(pin, GPIO.HIGH)
		for pin in ledPins[10:0:-1]:		#make led on from right to left
			GPIO.output(pin, GPIO.LOW)	
			time.sleep(0.1)
			GPIO.output(pin, GPIO.HIGH)

def destroy():
	for pin in ledPins:
		GPIO.output(pin, GPIO.HIGH)    # turn off all leds
	GPIO.cleanup()                     # Release resource

if __name__ == '__main__':     # Program start from here
	setup()
	try:
		loop()
	except KeyboardInterrupt:  # When 'Ctrl+C' is pressed, the program destroy() will be  executed.
		destroy()

